network.go

```diff
@@ -6,7 +6,10 @@ import (
 )
 
 func getHostByName(name string) string {
-	addrs, _ := net.LookupHost(name)
-	//TODO: add error handing when release v3 comes out
+	addrs, err := net.LookupHost(name)
```

Comment:
as this accesses an external resource a retry mechanism may be good (assuming it is not already builtin to net.LookupHost)